src/steps/command.ts

```diff
@@ -37,16 +37,22 @@ export class Command {
     toString(): string {
         return this.command;
     }
-}
 
-type Agents = Map<string, string>;
+    protected serialize(): string {
+        return this.toString();
+    }
 
-const transformCommand = (value: Command[]): undefined | string | string[] => {
-    if (!value || value.length === 0) {
-        return undefined;
+    static transformCommand(value: Command[]): undefined | string | string[] {
```

Comment:
Does this make sense as a static method @joscha? It felt strange to me since it accepts `Command[]`

Reply:
can you try if you can use `this[]`?

Reply:
you mean like `this.transformCommand`? Isn't that was `serialize()` does?

Reply:
I had a quick look, how do you envision this working? do you mean use `this[]` instead of passing in the list of commands? If so that doesn't work

Reply:
I meant https://www.typescriptlang.org/docs/handbook/2/classes.html#this-types - but it doesn't work on static functions, which makes sense as it couldn't derive the subclass type

Reply:
making the unexported `transformCommand` a `static transformCommand` on `Command` makes this internal API publicly available on `Command`. I think the best we can do right now is what `Half-Answer: Scoped Symbols` describes in https://stackoverflow.com/questions/22156326/private-properties-in-javascript-es6-classes - can you give it a try, please?